docs: changelog week 10 (#694)
TabbyML/pochi: packages/docs/content/docs/developer-updates/2025-11-13-weekly-update-10.mdx

---
title: Weekly Update #10
---

# Weekly Update #10

### TL;DR

We added modular `@import` rules and fixed PDF handling in `readFile`. Behind the scenes, we’re preparing two big improvements: smart task notifications and a refreshed, faster tab-completion experience.

### ✨ Enhancements

- **Modular Rules with `@import` Syntax:** You can now split and organize large rule files by importing other markdown files using the `@` prefix (e.g. `@./rules/api-guidelines.md`). Pochi will stitch them together, managing complex configurations for your multi-agent or multi-project setups. **[#540](https://github.com/TabbyML/pochi/issues/540)** 

<div
className="w-12/15 md:ml-8 *:shadow-xl/30 *:rounded-lg"
  style={{
    position: "relative",
    paddingBottom: "53.7%",
    marginBottom: 20,
    height: 0,
    overflow: "hidden",
    maxWidth: "100%",
  }}
>
  <iframe
    src="https://drive.google.com/file/d/18s3_3KyoA02aUefK7zmMJirWljW_QOFF/preview"
    style={{
      position: "absolute",
      top: 0,
      left: 0,
      width: "100%",
      height: "100%",
      border: "none",
      borderRadius: "8px",
    }}
    allowFullScreen
    title="Modular rules"
  />
</div>

### 🐛 Bug fixes

> - **PDF Handling in `readFile`:** Resolved a bug that caused `readFile` to return binary contents for PDF files. **[#591](https://github.com/TabbyML/pochi/issues/591)**

Reviewer: ```suggestion
- **PDF Handling in `readFile`:** Resolved a bug that caused `readFile` to return PDF files as plain text content. **[#591](https://github.com/TabbyML/pochi/issues/591)**
```
